sympy/concrete/summations.py

```diff
@@ -1194,3 +1285,53 @@ def eval_sum_hyper(f, i_a_b):
                     return S.NegativeInfinity
             return None
         return Piecewise(res, (old_sum, True))
+
+
+def _eval_matrix_sum(expression):
+    f = expression.function
+    for n, limit in enumerate(expression.limits):
+        i, a, b = limit
+        dif = b - a
+        if dif.is_Integer:
+            if (dif < 0) == True:
+                a, b = b + 1, a - 1
+                f = -f
+
+            newf = eval_sum_direct(f, (i, a, b))
+            if newf is not None:
+                return newf.doit()
+
+
+def _dummy_with_inherited_properties_concrete(limits):
+    """
+    Return a Dummy symbol that inherits as much assumptions based on the
+    provided symbol and limits as possible.
```

Comment:
What happens if the provided symbol has assumptions that contradict the limits?

Reply:
Hopefully it throws an inconsistent assumption error? Maybe should be tested.

Reply:
I tried with this PR:
```julia
In [7]: x = Symbol('x', zero=False)                                                                                               

In [8]: Sum(x, (x, 0, 3)).doit()                                                                                                  
Out[8]: 6

In [9]: x = Symbol('x', positive=True)                                                                                            

In [10]: Sum(x, (x, 0, 3)).doit()                                                                                                 
Out[10]: 6
```

Reply:
Try putting both to zero. It will not add a zero assumption otherwise. I would say that this is consistent with much of SymPy at the moment (also, it would prohibit integer assumptions and summations with `oo`...).

However, I would expect 
```
x = Symbol('x', negative=True) 
Sum(x, (x, 0, 3)).doit() 
```
to throw an error.

Reply:
I think that the assumptions on the symbol should be ignored. The symbol is a dummy that takes on values determined purely by the limits.

Reply:
The reason that it cannot be ignored (easily) is that it is called recursively and quits if no assumptions are added (similar design as for integrals). I also think it is good practice to put assumptions on variables, rather than relying on automatic assumptions.

Btw, 

```
In [7]: S(1)._assumptions
Out[7]: 
{'commutative': True,
 'integer': True,
 'algebraic': True,
 'noninteger': False,
 'irrational': False,
 'imaginary': False,
 'transcendental': False,
 'extended_real': True,
 'real': True,
 'rational': True,
 'infinite': False,
 'hermitian': True,
 'finite': True,
 'complex': True,
 'zero': False,
 'extended_nonzero': True,
 'nonzero': True}
```

So `negative` or `positive` are not set (except for `oo` etc). Seems to make sense to fix that issue either way.

Reply:
Strange:
```
Integer(1)._assumptions
Out[26]: 
{'commutative': True,
 'integer': True,
 'algebraic': True,
 'noninteger': False,
 'irrational': False,
 'imaginary': False,
 'transcendental': False,
 'extended_real': True,
 'real': True,
 'rational': True,
 'infinite': False,
 'hermitian': True,
 'finite': True,
 'complex': True,
 'zero': False,
 'extended_nonzero': True,
 'nonzero': True,
 'positive': True,
 'negative': False,
 'nonnegative': True,
 'nonpositive': False,
 'extended_nonpositive': False,
 'extended_positive': True,
 'extended_nonnegative': True,
 'extended_negative': False}
````

Reply:
Oh, and a reason to not use the assumptions on the variables is that then one is forced to set assumptions on symbolic bounds. In this way, it is enough to do it on either.

Reply:
That's why you shouldn't use `_assumptions` directly. It contains the assumptions that have already been checked. When you do `x.is_real` it will check `x._assumptions['real']` then call `x._eval_is_real` and then if that gives True/False update `x._assumptions`. So `_assumptions` is an internal cache.
https://github.com/sympy/sympy/blob/d7a854fb789591114d7d51c3f23bba2fd4d520cc/sympy/core/assumptions.py#L257-L269
If `_eval_is_real` returns None then `_ask` will go on to check any other combination of facts that could imply real e.g. `_eval_is_prime`, `_eval_is_positive`, `_eval_is_extended_real+_eval_is_finite` etc.

Reply:
> Oh, and a reason to not use the assumptions on the variables is that then one is forced to set assumptions on symbolic bounds. In this way, it is enough to do it on either.

I think that either:
1. It should be checked that the assumptions on both are not known to be inconsistent.
2. The assumptions on the symbol should be ignored.